src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,165 @@
+                remove(i);
+                return true;
+            }
+        }
+        throw new NoSuchElementException("Елемент не знайдено.");
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    public boolean contains(Object o) {
+        for (int i = 0; i < size; i++) {
+            if (elements[i].equals(o)) {
+                return true;
+            }
+        }
+        return false;
+    }
 
+    public void clear() {
+        for (int i = 0; i < size; i++) {
+            elements[i] = null;
+        }
+        size = 0;
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public T set(int index, T element) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Індекс " + index + " виходить за межі списку.");
+        }
+        T oldValue = (T) elements[index];
+        elements[index] = element;
+        return oldValue;
     }
 
     @Override
-    public void set(T value, int index) {
+    public void add(int index, T element) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Індекс " + index + " виходить за межі списку.");
+        }
+        ensureCapacity();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = element;
+        size++;
+    }
 
+    public int indexOf(Object o) {
+        for (int i = 0; i < size; i++) {
+            if (elements[i].equals(o)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public int lastIndexOf(Object o) {
+        for (int i = size - 1; i >= 0; i--) {
+            if (elements[i].equals(o)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public Object[] toArray() {
+        Object[] array = new Object[size];
+        System.arraycopy(elements, 0, array, 0, size);
+        return array;
     }
 
     @Override
-    public T remove(int index) {
-        return null;
+    public <T1> T1[] toArray(T1[] a) {
+        throw new UnsupportedOperationException("Method not supported");
     }
 
     @Override
-    public T remove(T element) {
-        return null;
+    public boolean containsAll(java.util.Collection<?> c) {
+        throw new UnsupportedOperationException("Method not supported");
     }
 
     @Override
-    public int size() {
-        return 0;
+    public boolean addAll(java.util.Collection<? extends T> c) {
+        for (T element : c) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
-    public boolean isEmpty() {
-        return false;
+    public boolean addAll(int index, java.util.Collection<? extends T> c) {
+        throw new UnsupportedOperationException("Method not supported");
```

Comment:
If the method is not supported, consider throwing UnsupportedOperationException without a message or with a more descriptive message.